workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/WorkflowViewStub.kt

```diff
@@ -216,10 +217,12 @@ public class WorkflowViewStub @JvmOverloads constructor(
   ) {
     val oldViewOrNull = showing.visualOrNull
 
-    showing.replaceWith(rendering, context, viewEnvironment) { newView, doFirstUpdate ->
+    val parent = actual.parent as? ViewGroup
+      ?: throw IllegalStateException("WorkflowViewStub must have a non-null ViewGroup parent")
 
-      val parent = actual.parent as? ViewGroup
-        ?: throw IllegalStateException("WorkflowViewStub must have a non-null ViewGroup parent")
+    showing.replaceWith(
+      rendering, AndroidContainer(parent), viewEnvironment
```

Comment:
@wugeorgeq This was what drove the work in #895. If you diff this file again `origin/main`, you'll see that we had lost the bit where the factory is given access to `parent`.